src/textual/widgets/_tabbed_content.py

```diff
@@ -70,6 +73,26 @@ def __init__(
         )
 
 
+class AwaitTabbedContent:
+    """An awaitable return by [`TabbedContent`][textual.widgets.TabbedContent] methods that modify the tabs."""
+
+    def __init__(self, *awaitables: AwaitMount | AwaitRemove) -> None:
+        """Initialise the awaitable.
+
+        Args:
+            *awaitables: The collection of awaitables to await.
+        """
+        super().__init__()
+        self._awaitables = awaitables
+
+    def __await__(self) -> Generator[None, None, None]:
+        async def await_tabbed_content() -> None:
+            for awaitable in self._awaitables:
+                await awaitable
+
+        return await_tabbed_content().__await__()
```

Comment:
I need to learn more about `__await__`.
Is it a "standard" approach to implement `__await__` with a wrapper around what we care about and then call `__await__` on the wrapper?